### PR TITLE
Add "Current Action" trigger for classic fixes #1628

### LINKED ISSUE #1628: Maul/Heroic strike visible toggle
**Is your feature request related to a problem? Please describe.**
I am unable to get druid maul (and warrior heroic strike) to visibly show that it has been queued/toggled in weakauras

**Describe the solution you'd like**
There should be something that looks for this state so that the weak aura icon can visibly change when queued/toggled

**Describe alternatives you've considered**
I tried many ways to get it to work but was unable to

### PR DESCRIPTION
**Pull request by mrbuds**
# Description

Add a new trigger "Current Action" for queued spells

img

## Type of change
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

Test on a warrior with heroic strike

before use: https://i.imgur.com/iCK1PYw.jpg
when used: https://i.imgur.com/0Pf4ewa.jpg
queued but no more rage: https://i.imgur.com/3aimgfl.jpg

### CODE REVIEW

File: WeakAuras/Prototypes.lua [hunk 1]
```diff
@@ -6215,6 +6215,47 @@ WeakAuras.event_prototypes = {
...
+      "ACTIONBAR_SLOT_CHANGED",
+      "ACTIONBAR_PAGE_CHANGED"
+    },
+    name = L["Current Action"],
+    init = function(trigger)
+      local _, _, icon, _, _, _, spellid = GetSpellInfo(trigger.spellName)
+      local button
+      if icon and spellid then
+          icon = icon
+          local slotList = C_ActionBar.FindSpellActionButtons(spellid)
```
Comment (emptyrivers) [inline WeakAuras/Prototypes.lua]: This is a neat API. Do you happen to know how it handles macros?

Comment (mrbuds) [reply WeakAuras/Prototypes.lua]: On my warrior i only have a macro for heroic strike in my action bars and it works

Comment (emptyrivers) [reply WeakAuras/Prototypes.lua]: Cool. We might be able to actually give people a solution when they want their auras to have keybind text.

File: WeakAuras/Prototypes.lua [hunk 2]
```diff
@@ -6215,6 +6215,47 @@ WeakAuras.event_prototypes = {
...
+          button = slotList[1]
```
Comment (emptyrivers) [inline WeakAuras/Prototypes.lua]: This value isn't used

File: WeakAuras/Prototypes.lua (hunk 1 above)
Comment (emptyrivers) [reply WeakAuras/Prototypes.lua]: Also, add C_ActionBar to .luacheckrc

File: WeakAuras/Prototypes.lua (hunk 2 above)
Comment (mrbuds) [reply WeakAuras/Prototypes.lua]: ho wait i uploaded wrong version

File: WeakAuras/Prototypes.lua [hunk 3]
```diff
@@ -6215,6 +6215,47 @@ WeakAuras.event_prototypes = {
     automaticrequired = true
   },
 
+  ["Current Action"] = {
```
Comment (InfusOnWoW) [inline WeakAuras/Prototypes.lua]: I think we need a better name.
Discord search is broken, but we should look up what terminology people are using when they ask for that functionality.

Comment (InfusOnWoW) [reply WeakAuras/Prototypes.lua]: It seems "queued" is the right word?

Comment (InfusOnWoW) [reply WeakAuras/Prototypes.lua]: Actually probably best to change this here too, as long as we can still do that.

(Note, also need to change the    WeakAuras.event_prototypes["Current Action"] = nil line)

**Comment by mrbuds** [comment]
I have added the removal of the totem trigger for classic, totem api was removed with last patch

**Pull request merged**